### PR TITLE
Convert src/annotator/sidebar.coffee to JS

### PR DESCRIPTION
**Pull request by robertknight**
Convert this module, but not its tests, to JS. This is a pretty
straightforward conversion with a few minor improvements:

 - The initialization of `options` was rewritten in a way that is
   consistent with `PdfSidebar` and avoids a non-idiomatic way of
   passing initialization parameters from derived to base class

 - Some comments that were hard to parse were rewritten or removed if
   they stated something obvious

 - Private methods and fields related to gesture state were prefixed
   with an underscore. Some of the sidebar tests did and still do access
   these, which is not ideal.

### CODE REVIEW

**Comment by robertknight** [comment]
The conversion of the existing code is done. This surface existing issues with code paths missing tests. I plan to test those code paths manually for the moment and add those once the existing tests are converted.

**Comment by robertknight** [comment]
In converting this I tested support for resizing the sidebar manually. The code also has support for swipe gestures to open and close the sidebar. I don't think these actually work at all in modern browsers because the swipe gesture instead moves back and forth in browser history. I converted the code as-is and will look at removing that logic separately.

File: src/annotator/sidebar.js [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+      // If using our own container frame for the sidebar, add the toolbar to it.
+      this.frame.prepend(toolbarContainer);
+      this.toolbarWidth = this.toolbar.getWidth();
+    } else {
+      // If using a host-page provided container for the sidebar, the toolbar is
+      // not shown.
+      this.toolbarWidth = 0;
+    }
+
+    this._gestureState = {
```
Comment (lyzadanger) [inline src/annotator/sidebar.js:66]: In scanning this, it's not clear why the values for the properties would be numbers. I'll keep reading the module and see if it becomes clearer...

Comment (lyzadanger) [reply src/annotator/sidebar.js:66]: Upon further reading, I believe these are pixel values. Perhaps a comment here?

File: src/annotator/sidebar.js [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+        if (width >= MIN_RESIZE) {
+          this.frame.css('width', `${width}px`);
+        }
+        this._notifyOfLayoutChange();
+      }
+    });
+  }
+
+  /**
+   * Notify integrator when sidebar is opened, closed or resized.
```
Comment (lyzadanger) [inline src/annotator/sidebar.js:165]: I noticed this in my own forays into the coffeescript variant of this module: what does _integrator_ refer to here?

File: src/annotator/sidebar.js [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+   *
+   * @param {boolean} [expanded] -
+   *   `true` or `false` if the sidebar is being directly opened or closed, as
+   *   opposed to being resized via the sidebar's drag handles
+   */
+  _notifyOfLayoutChange(expanded) {
+    if (!this.onLayoutChange) {
```
Comment (lyzadanger) [inline src/annotator/sidebar.js:172]: I like this refactored as a guard.

File: src/annotator/sidebar.js [hunk 4]
```diff
@@ -0,0 +1,326 @@
...
+  isOpen() {
+    if (this.frame) {
+      return !this.frame.hasClass('annotator-collapsed');
+    } else {
+      // Assume an external frame is always open.
+      return true;
+    }
+  }
+
+  setAllVisibleHighlights(shouldShowHighlights) {
```
Comment (lyzadanger) [inline src/annotator/sidebar.js:323]: What does this function accomplish (in a conceptual way)?

File: src/annotator/sidebar.js (hunk 2 above)
Comment (robertknight) [reply src/annotator/sidebar.js:165]: Here it means the party that embedded the client in the current document. I'm not sure that we have a standardized term for that. Embedder perhaps?

File: src/annotator/sidebar.js (hunk 4 above)
Comment (robertknight) [reply src/annotator/sidebar.js:331]: It turns the document highlights on or off. I've added a JSDoc comment.
